File: task.py
import time
import sys

sys.setrecursionlimit(1000000)
memory_limit = 1024 * 1024 * 1024
try:
    import resource
    resource.setrlimit(resource.RLIMIT_AS, (memory_limit, memory_limit))
except ImportError:
    pass

start_time = time.time()

file_path = '10m.txt'

numbers = []
try:
    with open(file_path, 'r') as file:
        for line in file:
            numbers.append(int(line))
except IOError:
    print("Не вдалося відкрити файл.")
    sys.exit()

def find_longest_increasing_subsequence(arr):
    longest = []
    current = []

    for num in arr:
        if len(current) == 0 or num > current[-1]:
            current.append(num)
        else:
            if len(current) > len(longest):
                longest = current
            current = [num]

    if len(current) > len(longest):
        longest = current

    return longest

def find_longest_decreasing_subsequence(arr):
    longest = []
    current = []

    for num in arr:
        if len(current) == 0 or num < current[-1]:
            current.append(num)
        else:
            if len(current) > len(longest):
                longest = current
            current = [num]

    if len(current) > len(longest):
        longest = current

    return longest

longest_increasing = find_longest_increasing_subsequence(numbers)
longest_decreasing = find_longest_decreasing_subsequence(numbers)

max_num = max(numbers)
min_num = min(numbers)

numbers.sort()
count = len(numbers)
median = (numbers[count // 2 - 1] + numbers[count // 2]) / 2 if count % 2 == 0 else numbers[count // 2]

average = sum(numbers) / count

end_time = time.time()
execution_time = end_time - start_time 

print(f"Максимальне число: {max_num}")
print(f"Мінімальне число: {min_num}")
print(f"Медіана: {median}")
print(f"Середнє арифметичне: {average}")
print(f"Найбільша зростаюча послідовність: {', '.join(map(str, longest_increasing))}")
print(f"Найбільша спадаюча послідовність: {', '.join(map(str, longest_decreasing))}")
print(f"Час обробки файлу: {execution_time} секунд")

File: task.php
<?php

ini_set('memory_limit', '1G');

$startTime = microtime(true);

$filePath = '10m.txt'; //вводимо шлях до текстового файлу розташованого локально на носії

$handle = fopen($filePath, "r");
if ($handle === false) {
    die("Не вдалося відкрити файл.");
}

$numbers = [];
while (($line = fgets($handle)) !== false) {
    $numbers[] = (int)$line;
}
fclose($handle);

$longestIncreasing = LongestIncreasingSubsequence($numbers);
$longestDecreasing = LongestDecreasingSubsequence($numbers);

// максимальне та мінімальне значення
$max = max($numbers);
$min = min($numbers);

// медіана
sort($numbers);
$count = count($numbers);
$median = ($count % 2 == 0) ? ($numbers[$count / 2 - 1] + $numbers[$count / 2]) / 2 : $numbers[floor($count / 2)];

// середнє арифметичне
$average = array_sum($numbers) / $count;

// найбільша послідовність чисел, яка збільшується
function LongestIncreasingSubsequence($arr) {
    $longest = [];
    $current = [];
    
    foreach ($arr as $i => $num) {
        if ($i === 0 || $num > end($current)) {
            $current[] = $num;
        } else {
            if (count($current) > count($longest)) {
                $longest = $current;
            }
            $current = [$num];
        }
    }
    if (count($current) > count($longest)) {
        $longest = $current;
    }
    return $longest;
}

//найбільша послідовність чисел, яка зменшується
function LongestDecreasingSubsequence($arr) {
    $longest = [];
    $current = [];
    
    foreach ($arr as $i => $num) {
        if ($i === 0 || $num < end($current)) {
            $current[] = $num;
        } else {
            if (count($current) > count($longest)) {
                $longest = $current;
            }
            $current = [$num];
        }
    }
    if (count($current) > count($longest)) {
        $longest = $current;
    }
    return $longest;
}



$endTime = microtime(true); // Кінець вимірювання часу
$executionTime = $endTime - $startTime; // Час виконання скрипта

echo "Максимальне число: $max\n";
echo "Мінімальне число: $min\n";
echo "Медіана: $median\n";
echo "Середнє арифметичне: $average\n";
echo "Найбільша зростаюча послідовність: " . implode(', ', $longestIncreasing) . "\n";
echo "Найбільша спадаюча послідовність: " . implode(', ', $longestDecreasing) . "\n";
echo "Час обробки файлу: $executionTime секунд\n";

?>
